M2/Macaulay2/packages/SuperLinearAlgebra.m2

```diff
@@ -0,0 +1,758 @@
+newPackage(
+    "SuperLinearAlgebra", 
+    Version => "0.1", 
+    Date => "29 January 2021", 
+    Authors => {
+        {Name => "Fereshteh Bahadorykhalily", 
+         Email => "f.bahadori.khalili@gmail.com", 
+         HomePage => "https://www.researchgate.net/profile/Fereshte_Bahadorykhalily"
+        }, 
+        {Name => "Fatemeh Tarashi Kashani", 
+         Email => "tarashikashanifatemeh@gmail.com", 
+         HomePage => "https://www.linkedin.com/in/fatemehtarashi/"
+        }
+    }, 
+    Headline => "berezinian and supertrace of a supermatrix", 
+    DebuggingMode => true
+)
+
+--------------------
+-- Exports
+--------------------
+
+export {
+   -- Types
+    "SuperMatrix", 
+    
+   -- Methods
+    "berezinian", 
+    "inverseSuperMatrix", 
+    "parity", 
+    "superMatrixGenerator", 
+    "superMatrixParity", 
+    "superRing", 
+    "superTrace", 
+    
+   -- Symbols
+    "supermatrix", 
+    "targetM1", 
+    "targetM3", 
+    "sourceM1", 
+    "sourceM2"
+}
+
+--------------------
+-- SuperRing (Super commutative ring) 
+--------------------
+
+superRing = method();
+superRing (PolynomialRing, PolynomialRing) := (R1, R2) -> (
+    inverseVariable := symbol inverseVariable;
+    R11 := (coefficientRing R1)[R1_0..R1_(#gens R1-1), inverseVariable_0..inverseVariable_(#gens R1-1)];
+    R11 = R11/apply(0..(#gens R1-1), i -> sub(R1_i, R11)*inverseVariable_i-1);
+    w := (for i to (#gens R2)-1 list (0))|toList(0..(#gens R2-1));
+    R22 := (coefficientRing R2)[R2_0..R2_(#gens R2-1), MonomialOrder=>{Weights => w, Lex}, SkewCommutative=>true];
+    print concatenate {"is a super commutative ring of dimension", toString (#gens R1), "|", toString (#gens R2)};
```

Comment:
Mathematical functions should generally not print, unless an error is being signaled.